scripts/sysman/PROG.rst

```diff
@@ -39,12 +45,51 @@ The following environment variables are required to be enabled during initializa
 Initialization
 --------------
 
+%if ver >= 1.5:
+An application wishing to manage power and performance for devices may
+use the System Resource Management library to enumerate system management
+driver and device handles.
+
+The following pseudo-code demonstrates a basic initialization and device discovery sequence:
+
+.. parsed-literal::
+
+   function main( ... )
+       if (${s}Init(0) != ${X}_RESULT_SUCCESS)
+           output("Can't initialize the API")
+       else
+           # Discover all the drivers
+           uint32_t driversCount = 0
+           ${s}DriverGet(&driversCount, nullptr)
+           ${s}_driver_handle_t* allDrivers = allocate(driversCount * sizeof(${s}_driver_handle_t))
+           ${s}DriverGet(&driversCount, allDrivers)
+
+           ${s}_driver_handle_t hDriver = nullptr
+           for(i = 0 .. driversCount-1)
+               # Discover devices in a driver
+               uint32_t deviceCount = 0
+               ${s}DeviceGet(allDrivers[i], &deviceCount, nullptr)
+
+               ${s}_device_handle_t* hSysmanHandles =
+                   allocate_memory(deviceCount * sizeof(${s}_device_handle_t))
+               ${s}DeviceGet(allDrivers[i], &deviceCount, hSysmanHandles)
+
+               # Use the hSymanHandles to manage the devices
+
+       free_memory(...)
+
+For compatibility, an application may also use the Level0 Core API to
+enumerate through available accelerator devices in the system. For
+each device handle, an application can cast it to a sysman device handle
```

Comment:
@bashbaug : can the opposite be done? cast a sysman device to a core device? this goes to the same as below, whether zesInit is enough or we need to call zeInit as well.

Reply:
No, the opposite should not be considered safe, even if may currently work in practice.

Reply:
thanks @bashbaug . So this aligns with what we said above about the affinity mask. the affinity mask would apply to Core, and therefore, the Core devices would be a subset of the sysman devices, so you could cast a core device to sysman one (you would always have one), but not the other way around, since you may have masked out the core device associated for the sysman device.